virttest/data_plane.py

```diff
@@ -0,0 +1,398 @@
+"""
+Autotest implementation of iothread manager classes.
+
+These classes represents different iothread allocation scheme.
+"""
+import heapq
+import re
+
+from avocado.utils import cpu
+from virttest.qemu_devices.qdevices import IOThread
+
+
+class MinQueue(object):
+    """Min heap."""
+
+    def __init__(self, item_list=None):
+        """Heapify the pool."""
+        self._queue = list(item_list or [])
+        heapq.heapify(self._queue)
+
+    def __len__(self):
+        """Return the length."""
+        return len(self._queue)
+
+    def __iter__(self):
+        """Return iterator."""
+        return iter(self._queue)
+
+    def __getitem__(self, pos):
+        """heap.__getitem__(pos) <==> heap[pos]"""
+        return self._queue[pos]
+
+    def __setitem__(self, pos, value):
+        """heap.__setitem__(pos, value) <==> heap[pos] = value"""
+        self._queue[pos] = value
+
+    def siftup(self, pos):
+        """Rearrange if item on pos is increased in value."""
+        heapq._siftup(self._queue, pos)
```

Comment:
I am not sure why you need to use `_method`. It isn't in the documentation. It is an internal implementation detail.
Once you `heapify` the queue, it does not change anything based on my unit test.

Reply:
Exposing those interfaces is to avoid re-heapify the heap if an item is changed in value, which could improve from `O(N) ` to `O(log(N))`.